### PR TITLE
make one file that handles bumping verisons

### PR DESCRIPTION
**Pull request by crookse**
Running the tests (make sure you have `use-cli-services` branch checked out):

```
deno install -A -f --reload /path/to/your/rhum/cli.ts
```

Example usage:

```typescript
import { BumperService } from "https://raw.githubusercontent.com/drashland/services/master/ci/bumper_service.ts";

const b = new BumperService("rhum", Deno.args);

if (BumperService.isForPreRelease) {
  b.bump([
    {
      filename: "./README.md",
      replaceTheRegex: /dmm@v[0-9\.]+[0-9\.]+[0-9\.]/g,
      replaceWith: `dmm@v{{ thisModulesLatestVersion }}`,
    },
    {
      filename: "./egg.json",
      replaceTheRegex: /"version": "[0-9\.]+[0-9\.]+[0-9\.]"/,
      replaceWith: `"version": "{{ thisModulesLatestVersion }}"`,
    },
    {
      filename: "./mod.ts",
      replaceTheRegex: /version = ".+"/,
      replaceWith: `version = "{{ thisModulesLatestVersion }}"`,
    },
    {
      filename: "./src/cli/commands/make.ts",
      replaceTheRegex:
        /import \{ Rhum \} from "https\:\/\/deno.land\/x\/rhum\@v[0-0]\.[0-9]\.[0-9]\/mod.ts"/g,
      replaceWith:
        `import { Rhum } from "https://deno.land/x/rhum@v{{ thisModulesLatestVersion }}/mod.ts"`,
    },
    {
      filename: "./src/commands/version.ts",
      replaceTheRegex: /version = "[0-9\.]+[0-9\.]+[0-9\.]"/,
      replaceWith: `version = "v{{ thisModulesLatestVersion }}"`,
    },
  ]);
} else {
  b.bump([
    {
      filename: "./.github/workflows/master.yml",
      replaceTheRegex: /deno: ["[0-9.]+[0-9.]+[0-9]"]/g,
      replaceWith: `deno: ["{{ latestDenoVersion }}"]`,
    },
    {
      filename: "./.github/workflows/bumper.yml",
      replaceTheRegex: /deno: ["[0-9.]+[0-9.]+[0-9]"]/g,
      replaceWith: `deno: ["{{ latestDenoVersion }}"]`,
    },
    {
      filename: "./.github/workflows/pre_release.yml",
      replaceTheRegex: /deno: ["[0-9.]+[0-9.]+[0-9]"]/g,
      replaceWith: `deno: ["{{ latestDenoVersion }}"]`,
    },
  ]);
}

```

### CODE REVIEW

**Review by ebebbington — APPROVED** [review]
LGTM

File: ci/bumper_service.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+   *
+   * @returns An object of key-value pairs where the key is the module in
+   * question and the value is the module's latest version.
+   */
+  public async getLatestVersions(): Promise<{ [key: string]: string }> {
+    return {
+      [this.module_name]: await this.getModulesLatestVersion(this.module_name),
+      deno: await this.getModulesLatestVersion("deno"),
+      deno_std: await this.getModulesLatestVersion("std"),
+    };
```
Comment (ebebbington) [inline ci/bumper_service.ts:185]: ooo sexy i like it

File: ci/bumper_service.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+   *   - Replace std versions; use `...: "{{ latestStdVersion }}"`
+   *   - Replace deno versions; use `...: "{{ latestDenoVersion }}"`
+   *
+   * All versions do not have the `v`, so add them yourself
+   *
+   * @param files - Files to replace content in and re-write to fs
+   * @param write - Should this method write to the filesystem? Defaults to
+   * true.
+   */
+  public async bump(files: File[], write: boolean = true): Promise<string[]> {
```
Comment (ebebbington) [inline ci/bumper_service.ts:106]: method doesn't seem to replace drash versions? or is that not a thing in this method anymore?

File: tests/ci/bumper_service_test.ts [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+        deno_std: await c.getModulesLatestVersion("std"),
+      };
+      Rhum.asserts.assertEquals(
+        await c.getLatestVersions(),
+        versions,
+      );
+    });
+  });
+
+  await Rhum.testSuite("bump", async () => {
```
Comment (ebebbington) [inline tests/ci/bumper_service_test.ts:50]: awaits?

File: ci/bumper_service.ts (hunk 2 above)
Comment (crookse) [reply ci/bumper_service.ts:106]: is there a reason why it should? i was under the impression that the bump for pre-release is the one that handles updating drash versions.

File: tests/ci/bumper_service_test.ts (hunk 3 above)
Comment (crookse) [reply tests/ci/bumper_service_test.ts:50]: yea there was an issue with the test not waiting for the fetch calls to finish. so the subsequent test suite would run while this one was still running.

Comment (crookse) [reply tests/ci/bumper_service_test.ts:50]: hmm... maybe the issue is with the last test suite and not this one. hldn. let me investigate.

Comment (crookse) [reply tests/ci/bumper_service_test.ts:50]: yea.. without these awaits, the tests run concurrently

**Review by ebebbington — APPROVED** [review]
Use of await makes sense, ignore my comment about the drash version, dmm will handle that separately

**Pull request merged**